sdk/dotnet/Pulumi.Automation/Exceptions/ProjectSettingsConflictException.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016-2021, Pulumi Corporation
+
+using System;
+
+namespace Pulumi.Automation.Exceptions
+{
+    /// <summary>
+    ///
+    /// Thrown when creating a Workspace detects a conflict between
+    /// project settings found on disk (such as Pulumi.yaml) and a
+    /// ProjectSettings object passed to the Create API.
+    ///
+    /// There are two resolutions:
+    ///
+    /// (A) to use the ProjectSettings, delete the Pulumi.yaml file
+    ///     from WorkDir or use a different WorkDir
+    ///
+    /// (B) to use the exiting Pulumi.yaml from WorkDir, avoid
+    ///     customizing the ProjectSettings
+    ///
+    /// </summary>
+    public class ProjectSettingsConflictException : Exception
+    {
+        internal string SettingsFileLocation { get; }
```

Comment:
I would make this public, no reason for anything to be `internal` other than the constructor I think. If a consumer catches it I don't think there is a reason for us to hide this path.

Reply:
I think that's what we have now! Thank you.